File: nodejs/code-duplication/good/taxCalculator.js
class TaxCalculator {
    static TAX_RATES = {
        'CA': 0.0825,
        'NY': 0.04,
        'TX': 0.0625,
    };

    calculateTax(price, state) {
        const taxRate = TaxCalculator.TAX_RATES[state] || 0.0;
        return price * taxRate;
    }
}

class ShippingCalculator {
    static BASE_RATE = 5.0;
    static WEIGHT_RATE_PER_UNIT = 0.5;
    static DISTANCE_RATE_PER_UNIT = 0.1;

    calculateShipping(weight, distance) {
        const weightRate = weight * ShippingCalculator.WEIGHT_RATE_PER_UNIT;
        const distanceRate = distance * ShippingCalculator.DISTANCE_RATE_PER_UNIT;

        return ShippingCalculator.BASE_RATE + weightRate + distanceRate;
    }
}

module.exports = {
    TaxCalculator,
    ShippingCalculator
};

File: csharp/code-duplication/good/TaxCalculator.cs
using System.Collections.Generic;

public class TaxCalculator
{
    private static readonly Dictionary<string, double> TaxRates = new Dictionary<string, double>
    {
        ["CA"] = 0.0825,
        ["NY"] = 0.04,
        ["TX"] = 0.0625
    };

    public double CalculateTax(double price, string state)
    {
        double taxRate = TaxRates.ContainsKey(state) ? TaxRates[state] : 0.0;
        return price * taxRate;
    }
}

public class ShippingCalculator
{
    private const double BaseRate = 5.0;
    private const double WeightRatePerUnit = 0.5;
    private const double DistanceRatePerUnit = 0.1;

    public double CalculateShipping(double weight, double distance)
    {
        double weightRate = weight * WeightRatePerUnit;
        double distanceRate = distance * DistanceRatePerUnit;

        return BaseRate + weightRate + distanceRate;
    }
}
